flow/connectors/utils/aws.go

```diff
@@ -178,6 +201,20 @@ func GetAWSCredentialsProvider(ctx context.Context, connectorName string, peerCr
 	return NewConfigBasedAWSCredentialsProvider(awsConfig), nil
 }
 
+func FileURLForS3Service(endpoint string, region string, bucket string, filePath string) string {
+	fileUrl := fmt.Sprintf("https://%s.s3.%s.amazonaws.com/%s", bucket, region, filePath)
+	if strings.Contains(endpoint, "storage.googleapis.com") {
+		fileUrl = fmt.Sprintf("https://storage.googleapis.com/%s/%s", bucket, filePath)
+	}
+	// example: min.io local bucket or GCS
+	match, _ := regexp.MatchString(`^https?://[a-zA-Z0-9.-]+(:\d+)?$`, endpoint)
```

Comment:
move to a global `var x = regexp.MustCompile("^https?://[a-zA-Z0-9.-]+(:\d+)?$")`